lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -64,8 +69,43 @@ def _on_pbm_config_changed(self, event) -> None:
             event.defer()
             return
 
-        # TODO handle PBM configurations set by the user
-        self.charm.unit.status = ActiveStatus()
+        # URI is set with `snap set`
+        pbm_snap.set({"uri": self._backup_config.uri})
+
+        # presets for PBM snap configurations
+        pbm_configs = {}
+        pbm_configs["storage.type"] = "s3"
+        pbm_configs["storage.s3.serverSideEncryption.sseAlgorithm"] = "aws:kms"
+
+        credentials = self.s3_client.get_s3_connection_info()
+        for s3_option, s3_value in credentials.items():
+            if s3_option not in S3_PBM_OPTION_MAP:
+                continue
+
+            pbm_name = S3_PBM_OPTION_MAP[s3_option]
+            pbm_configs[pbm_name] = s3_value
+
+        for (pbm_key, pbm_value) in pbm_configs.items():
+            try:
+                self._pbm_set_config(pbm_key, pbm_value)
+            except subprocess.CalledProcessError as e:
```

Comment:
maybe in the moment of rotation of credentials, we always will get this exception.
because the new `access-key` will be set with  the old `secret-key` in place.

Reply:
from what I can tell this does not fail when changing the `access-key`

Reply:
:pensive: unfortunately the command `percona-backup-mongodb config --set` can only set **one option at a time** so it is impossible to set all the options at one time with `config --set`. 

I could write them to a file in the `percona-backup-mongodb` snap directory and then do `percona-backup-mongodb config --file=<file in snap direction with desired options>` :thinking_face: